### ISSUE TITLE
Some Qobuz music not playing on Sonos

### ISSUE THREAD

**Issue by milkplus27** [music-assistant/hass-music-assistant#1823]
### What version of Music Assistant has the issue?

 2024.1.0

### The problem

Hey there. I'm going a little nuts... until today I was able to play music from Qobuz on my Sonos speakers via Music Assistant with no issues. But now anything i try to play that is marked as HiRes seems to not work. I understand that Sonos speakers don't support HiRes, but i thought that then it would automatically play the next possible version.

Albums that previously worked ok now just won't play at all. I also seem to remember there was a setting somewhere where you can set the highest playback quality for a player but i can't for the life of me find it... was it removed in an update?

Not sure if others have encountered this problem?

### How to reproduce

Select a HiRes track from Qobuz and attempt to play it on a Sonos speaker

### What version of Home Assistant Core are your running

2024.1.5

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by OzGav** [comment]
Yes the codec setting was removed in he last update. What about logs?

**Comment by milkplus27** [comment]
yes, of course, sorry. here you go. this is what shows up in the logs when i try to play a HiRes Qobuz file on my Kitchen Play 1.

```
2024-01-24 06:29:15.140 DEBUG (MainThread) [music_assistant.webserver] [139845162029776] Handling command music/item_by_uri
2024-01-24 06:29:17.558 DEBUG (MainThread) [music_assistant.webserver] [139845162029776] Handling command players/queue/play_media
2024-01-24 06:29:17.588 DEBUG (MainThread) [music_assistant.event] queue_items_updated RINCON_B8E93752747A01400
2024-01-24 06:29:17.592 DEBUG (MainThread) [music_assistant.event] queue_updated RINCON_B8E93752747A01400
2024-01-24 06:29:17.628 DEBUG (MainThread) [music_assistant.webserver] [139846360429840] Handling command music/item_by_uri
2024-01-24 06:29:17.639 DEBUG (MainThread) [music_assistant.webserver] [139845162029776] Handling command music/item_by_uri
2024-01-24 06:29:17.681 DEBUG (MainThread) [music_assistant.providers.sonos] Enqued track (The Mountain Goats - Clean Slate) to player Kitchen Sonos
2024-01-24 06:29:17.698 DEBUG (Thread-25 (process_request_thread)) [music_assistant.providers.sonos] Received AVTransport event for Player Kitchen
2024-01-24 06:29:17.700 DEBUG (MainThread) [music_assistant.event] queue_updated RINCON_B8E93752747A01400
2024-01-24 06:29:17.701 DEBUG (MainThread) [music_assistant.event] player_updated RINCON_B8E93752747A01400
2024-01-24 06:29:17.722 DEBUG (MainThread) [music_assistant.webserver] [139846360429840] Handling command music/item_by_uri
2024-01-24 06:29:17.737 DEBUG (MainThread) [music_assistant.webserver] [139845162029776] Handling command music/item_by_uri
2024-01-24 06:29:17.771 DEBUG (MainThread) [music_assistant.streams] Got GET request to /RINCON_B8E93752747A01400/single/72ae014fcb034474adb9353699cbbada.flac from 192.168.1.92
headers: <CIMultiDictProxy('Connection': 'close', 'Host': '192.168.1.155:8096', 'Accept': '*/*', 'User-Agent': 'Linux UPnP/1.0 Sonos/76.2-47270 (ZPS1)', 'X-Sonos-SWGen': '2', 'X-Sonos-Firmware': '76.2-47270', 'X-Sonos-Muse-Api': '1.37.1', 'X-Sonos-Id-Hash': 'g+eNtcZyZT7mhE4NVZFQUp4HWR9V+GelGhczpDohSy8=')>

2024-01-24 06:29:17.902 DEBUG (Thread-26 (process_request_thread)) [music_assistant.providers.sonos] Received AVTransport event for Player Kitchen
2024-01-24 06:29:17.977 DEBUG (MainThread) [music_assistant.streams] Start serving audio stream for QueueItem qobuz://track/216020856 to Kitchen Sonos
2024-01-24 06:29:17.999 DEBUG (MainThread) [music_assistant.audio] start media stream for: qobuz://track/216020856
2024-01-24 06:29:18.390 DEBUG (MainThread) [music_assistant.streams] Got HEAD request to /RINCON_B8E93752747A01400/single/72ae014fcb034474adb9353699cbbada.flac from 192.168.1.92
headers: <CIMultiDictProxy('Connection': 'close', 'Host': '192.168.1.155:8096', 'User-Agent': 'Linux UPnP/1.0 Sonos/76.2-47270 (ZPS1)', 'X-Sonos-SWGen': '2', 'X-Sonos-Firmware': '76.2-47270', 'X-Sonos-Muse-Api': '1.37.1', 'X-Sonos-Id-Hash': 'g+eNtcZyZT7mhE4NVZFQUp4HWR9V+GelGhczpDohSy8=')>

2024-01-24 06:29:18.390 DEBUG (MainThread) [music_assistant.audio] Using cached streamdetails for qobuz://track/216020856
2024-01-24 06:29:18.580 WARNING (MainThread) [asyncio] Executing <Task finished name='Task-2676' coro=<RequestHandler._handle_request() done, defined at /usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py:442> result=(<StreamResponse OK eof>, False) created at /usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py:527> took 0.191 seconds
2024-01-24 06:29:18.694 DEBUG (Thread-27 (process_request_thread)) [music_assistant.providers.sonos] Received AVTransport event for Player Kitchen
2024-01-24 06:29:19.714 DEBUG (MainThread) [music_assistant.audio] stripped 3.97 seconds of silence from end of pcm audio. bytes stripped: 2284428
2024-01-24 06:29:19.715 DEBUG (MainThread) [music_assistant.audio] finished media stream for: qobuz://track/216020856
2024-01-24 06:29:19.723 DEBUG (MainThread) [music_assistant.audio] Start analyzing audio for qobuz://track/216020856
2024-01-24 06:29:20.968 DEBUG (MainThread) [music_assistant.audio] Integrated loudness of qobuz://track/216020856 is: -8.1
```

**Comment by milkplus27** [comment]
Also, if i then go into the Sonos app and try to press play on that speaker (which shows the track loaded in) it gives me the following error message:

`Unable to play 'Clean Slate' - It is encoded at unsupported sample rate  24b96000 hz`

**Comment by milkplus27** [comment]
i uninstalled and reinstalled the add-on + integration (both latest versions) and that seems to have fixed the issue for now.

But i think it might have something to do with grouping the players. I seem to recall that it started happening when i had grouped two players together (Sonos Arc and Play:1) and then music did not play and i had ungrouped and things stopped working from there. Will try and see if I can recreate it...

**Comment by milkplus27** [comment]
hmm, yes... it really seems like it's the grouping that's causing issues. when i try to group players together (all Sonos) MA starts behaving strangely. it will play HiRes files on the Arc but not the Play:1 and then doesn't let me ungroup the players even after restarting Music Assistant/Home Assistant. They are ungrouped in Sonos, but remain grouped in MA. The only way to fix seems to uninstall and reinstall.

**Comment by OzGav** [comment]
Ok good fault finding. I will add this to the backlog so get looked at soon.

**Comment by marcelveldt** [comment]
hmmm, yeah a play:1 is gen1 hardware which does not support hires where s2 hardware supports up to 48/24 - somehow our entire check fails when playing to a group with both s1 and s2 hardware, I will look into it.

**Comment by OzGav** [comment]
This should be fixed in beta 86 please advise if not

**Comment by milkplus27** [comment]
Thanks! It seems that hires tracks from qobuz are playing fine on individual Sonos players, however I now am unable to group players together to test out if it breaks things like before. 

When I try to group two or more players together it just stops the music and puts all players on idle - unless the Arc is outputting TV audio, in which case it continues and other players stop. But nothing gets grouped either way.

**Comment by OzGav** [comment]
Please open a new issue don’t just add to an unrelated closed one.

**Comment by milkplus27** [comment]
Hmm played around with it some more and I'm sorry to report that it still seems broken. I created a group players of Play:1 and Arc which seems to have broken things again. Group won't play any music, and also now can't play anything on individual players again - this now appears to be regardless of whether the file is hires or not.

Looking at the Sonos app, it's spitting out an error msg (see attached) and trying to play music on the speaker group (even though the speakers are not groupedin the Sonos app) even if I tell MA to only play on one speaker.

More logs:

```
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/soco/core.py", line 849, in stop
    self.avTransport.Stop([("InstanceID", 0), ("Speed", 1)])
  File "/usr/local/lib/python3.11/site-packages/soco/services.py", line 207, in _dispatcher
    return self.send_command(action, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/soco/services.py", line 515, in send_command
    self.handle_upnp_error(response.text)
  File "/usr/local/lib/python3.11/site-packages/soco/services.py", line 571, in handle_upnp_error
    raise SoCoUPnPException(
soco.exceptions.SoCoUPnPException: UPnP Error 701 received: Transition not available from 192.168.1.247
2024-01-27 08:10:15.545 ERROR (MainThread) [music_assistant.webserver] [140629247423312] Error handling message: CommandMessage(message_id=36, command='players/cmd/sync', args={'player_id': 'RINCON_B8E93752747A01400', 'target_player': 'RINCON_48A6B8B07BFF01400'})
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/webserver.py", line 372, in _run_handler
    result = await result
             ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/players.py", line 74, in wrapper
    await func(self, *args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/players.py", line 671, in cmd_sync
    await player_provider.cmd_sync(player_id, target_player)
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/providers/sonos/__init__.py", line 518, in cmd_sync
    sonos_player.soco_device.join, self.sonosplayers[target_player].soco
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'SonosPlayer' object has no attribute 'soco'
2024-01-27 08:14:48.216 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (Gerald Wilson Big Band - Teri) to player Kitchen
2024-01-27 08:15:11.860 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (Gerald Wilson Big Band - Nancy Jo) to player Kitchen
2024-01-27 08:17:40.917 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (Gerald Wilson Big Band - Nancy Jo) to player Kitchen
2024-01-27 08:18:04.437 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (Gerald Wilson Big Band - Milestones) to player Kitchen
2024-01-27 08:20:16.914 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (Gerald Wilson Big Band - Milestones) to player Kitchen
2024-01-27 08:20:37.152 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (Gerald Wilson Big Band - Latino) to player Kitchen
2024-01-27 08:22:00.090 ERROR (MainThread) [music_assistant.webserver] [140629650476176] Error handling message: CommandMessage(message_id=16, command='players/cmd/sync', args={'player_id': 'RINCON_B8E93752747A01400', 'target_player': 'RINCON_48A6B8B07BFF01400'})
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/webserver.py", line 372, in _run_handler
    result = await result
             ^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/players.py", line 74, in wrapper
    await func(self, *args, **kwargs)
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/controllers/players.py", line 671, in cmd_sync
    await player_provider.cmd_sync(player_id, target_player)
  File "/usr/local/lib/python3.11/site-packages/music_assistant/server/providers/sonos/__init__.py", line 518, in cmd_sync
    sonos_player.soco_device.join, self.sonosplayers[target_player].soco
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'SonosPlayer' object has no attribute 'soco'
2024-01-27 08:22:34.168 INFO (MainThread) [music_assistant.players] Player registered: ugp_ljcmlmdf/Kitchen + Living Room 
2024-01-27 08:23:08.026 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:23:08.060 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:23:13.127 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_B8E93752747A01400/command/next.mp3) to player Kitchen
2024-01-27 08:23:18.049 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_48A6B8B07BFF01400/command/next.mp3) to player Living Room
2024-01-27 08:23:18.051 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Kitchen + Living Room  - crossfade: False
2024-01-27 08:23:18.053 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:23:18.084 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:23:19.765 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue Kitchen + Living Room 
2024-01-27 08:24:33.989 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:24:34.054 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:24:39.116 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_B8E93752747A01400/command/next.mp3) to player Kitchen
2024-01-27 08:24:44.012 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_48A6B8B07BFF01400/command/next.mp3) to player Living Room
2024-01-27 08:24:44.013 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Kitchen + Living Room  - crossfade: False
2024-01-27 08:24:44.033 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:24:44.136 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:24:45.494 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue Kitchen + Living Room 
2024-01-27 08:24:57.044 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:24:57.090 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:25:02.148 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_B8E93752747A01400/command/next.mp3) to player Kitchen
2024-01-27 08:25:07.056 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_48A6B8B07BFF01400/command/next.mp3) to player Living Room
2024-01-27 08:25:07.057 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Kitchen + Living Room  - crossfade: False
2024-01-27 08:25:07.061 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:25:07.259 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:25:08.672 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue Kitchen + Living Room 
2024-01-27 08:26:04.850 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:04.918 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:09.981 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_B8E93752747A01400/command/next.mp3) to player Kitchen
2024-01-27 08:26:14.871 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_48A6B8B07BFF01400/command/next.mp3) to player Living Room
2024-01-27 08:26:14.872 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Kitchen + Living Room  - crossfade: False
2024-01-27 08:26:14.879 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:14.908 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:16.185 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue Kitchen + Living Room 
2024-01-27 08:26:40.512 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:40.556 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:45.607 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_B8E93752747A01400/command/next.mp3) to player Kitchen
2024-01-27 08:26:50.534 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_48A6B8B07BFF01400/command/next.mp3) to player Living Room
2024-01-27 08:26:50.536 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Kitchen + Living Room  - crossfade: False
2024-01-27 08:26:50.539 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:50.825 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:26:53.056 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue Kitchen + Living Room 
2024-01-27 08:30:48.808 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:30:48.855 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:30:53.923 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_B8E93752747A01400/command/next.mp3) to player Kitchen
2024-01-27 08:30:58.831 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_48A6B8B07BFF01400/command/next.mp3) to player Living Room
2024-01-27 08:30:58.833 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Kitchen + Living Room  - crossfade: False
2024-01-27 08:30:58.835 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:30:59.058 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:31:00.204 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue Kitchen + Living Room 
2024-01-27 08:32:01.538 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:32:01.581 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:32:06.654 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_B8E93752747A01400/command/next.mp3) to player Kitchen
2024-01-27 08:32:11.562 INFO (MainThread) [music_assistant.providers.sonos] Enqued next track (http://192.168.1.155:8096/RINCON_48A6B8B07BFF01400/command/next.mp3) to player Living Room
2024-01-27 08:32:11.564 INFO (MainThread) [music_assistant.streams] Start Queue Flow stream for Queue Kitchen + Living Room  - crossfade: False
2024-01-27 08:32:11.565 WARNING (MainThread) [music_assistant.streams] Player RINCON_48A6B8B07BFF01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:32:11.821 WARNING (MainThread) [music_assistant.streams] Player RINCON_B8E93752747A01400 is making multiple requests to the same stream, playback may be disturbed!
2024-01-27 08:32:12.627 INFO (MainThread) [music_assistant.streams] Finished Queue Flow stream for Queue Kitchen + Living Room
![Screenshot_20240127-083157](https://github.com/music-assistant/hass-music-assistant/assets/20435315/2325ae5e-a64e-41b3-97f3-38cb195c33f0)

```

**Comment by milkplus27** [comment]
> Please open a new issue don’t just add to an unrelated closed one.

Sorry. Will do :)
